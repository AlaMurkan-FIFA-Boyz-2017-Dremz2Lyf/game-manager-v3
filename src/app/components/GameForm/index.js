import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Header, Button, Modal } from 'semantic-ui-react';
import TeamSearch from '../TeamSearch';
import FinishGame from '../FinishGame';

class GameForm extends PureComponent {
  state = {
    modalOpen: false,
  }

  closeModal = () => {
    this.setState({
      modalOpen: false,
    });
  }

  openModal = () => {
    this.setState({
      modalOpen: true,
    });
  }

  render() {
    const { modalOpen } = this.state;
    const { status } = this.props;
    let headerContent = 'Select Teams';
    let buttonContent = 'Start';
    let buttonPrimary = false;

    if (status === 'started') {
      headerContent = 'Complete Game';
      buttonContent = 'Complete Game';
      buttonPrimary = true;
    }

    return (
      <Modal
        open={modalOpen}
        onOpen={this.openModal}
        onClose={this.closeModal}
        trigger={<Button primary={buttonPrimary} fluid>{buttonContent}</Button>}
        basic
        size="small"
      >
        <Header icon="soccer" content={headerContent} />
        {
          status === 'started' ? (
            <FinishGame closeModal={this.closeModal} />
          ) : (
            <TeamSearch closeModal={this.closeModal} />
          )
        }
      </Modal>
    );
  }
}
GameForm.propTypes = {
  status: PropTypes.string,
};

export default GameForm;
