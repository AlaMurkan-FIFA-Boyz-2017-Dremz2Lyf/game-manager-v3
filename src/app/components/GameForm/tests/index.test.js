import { shallow } from 'enzyme';
import React from 'react';

import GameForm from '../index';

describe('<GameForm />', () => {
  it('should match snapshot for started games', () => {
    const wrapper = shallow(<GameForm status="started" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot for other games', () => {
    const wrapper = shallow(<GameForm status="created" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a closeModal method', () => {
    const wrapper = shallow(<GameForm />);
    wrapper.setState({ modelOpen: true });
    expect(wrapper.state({ modelOpen: true }));
    wrapper.instance().closeModal();
    expect(wrapper.state({ modelOpen: false }));
  });

  it('should have a openModal method', () => {
    const wrapper = shallow(<GameForm />);
    expect(wrapper.state({ modelOpen: false }));
    wrapper.instance().openModal();
    expect(wrapper.state({ modelOpen: true }));
  });
});
