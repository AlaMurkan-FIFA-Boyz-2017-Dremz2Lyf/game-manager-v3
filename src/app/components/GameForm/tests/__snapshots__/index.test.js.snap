// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameForm /> should match snapshot for other games 1`] = `
<Modal
  basic={true}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  onClose={[Function]}
  onOpen={[Function]}
  open={false}
  size="small"
  trigger={
    <Button
      as="button"
      fluid={true}
      primary={false}
    >
      Start
    </Button>
  }
>
  <Header
    content="Select Teams"
    icon="soccer"
  />
  <TeamSearch
    closeModal={[Function]}
    getCompetitions={[Function]}
    getTeams={[Function]}
  />
</Modal>
`;

exports[`<GameForm /> should match snapshot for started games 1`] = `
<Modal
  basic={true}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  onClose={[Function]}
  onOpen={[Function]}
  open={false}
  size="small"
  trigger={
    <Button
      as="button"
      fluid={true}
      primary={true}
    >
      Complete Game
    </Button>
  }
>
  <Header
    content="Complete Game"
    icon="soccer"
  />
  <FinishGame
    closeModal={[Function]}
    imageFetcher={[Function]}
  />
</Modal>
`;
